src/main/logseq/api.cljs

```diff
@@ -477,11 +496,13 @@
                     before?)
           new-block (editor-handler/api-insert-new-block!
                       content
-                      {:block-uuid block-uuid'
-                       :sibling?   sibling?
-                       :before?    before?
-                       :page       page-name
-                       :properties properties})]
+                      {:block-uuid  block-uuid'
+                       :sibling?    sibling?
+                       :before?     before?
+                       :page        page-name
+                       :custom-uuid custom-uuid
+                       :properties  (merge properties
+                                           (when custom-uuid {:id custom-uuid}))})]
```

Comment:
@sawhney17 @xyhp915 could we support this for pages too? Pages currently don't even have the `upsertBlockProperty(page, id, uuid)` workaround since it seems like `upsertBlockProperty` only works on blocks and not pages.

Basically it looks like you could only add page properties on init

Reply:
Oh looks like I could add a property after the fact by simply creating a block with `id::`

It seems like `PageEntity` should be updated to support `properties` field

Reply:
> Oh looks like I could add a property after the fact by simply creating a block with `id::`
> 
> It seems like `PageEntity` typescript definition should be updated to support `properties` field, since the API does return it for pages with properties

Thanks. Will be fixed soon :)